### ISSUE TITLE
Cannot create a pooled connection. "The trait bound is not satisfied"

### ISSUE THREAD

**Issue by mpiorowski** [weiznich/diesel_async#78]
### Versions

- **Rust:** -  rustc 1.62.1
- **Diesel:** - 2.0.4
- **Diesel_async:** - 0.2.2
- **Database:** - postgres
- **Operating System** - linux

### Feature Flags

- **diesel:** - ["time", "uuid", "numeric"]
- **diesel_async:** - ["postgres", "deadpool", "bb8"]

## Problem Description
Cannot created a pooled connection, and pass it to tonic. While trying to specify the type, rust analyzer throws an error.

### What are you trying to accomplish?
I am trying to created a pooled diesel-async connection (any really, tried every example, now using the newest one on this github with rustls). Everythime i am trying to specify the type, compalier throws an error, that i am mssing some trait implementation.

![image](https://github.com/weiznich/diesel_async/assets/26543876/eb7e092e-8ec2-423b-8fd7-541ac1a460e6)

## Github example
https://github.com/mpiorowski/rusve/blob/diesel/service-notes/src/main.rs

**Comment by mpiorowski** [comment]
I dont know if this will help You @weiznich but when i try passing the `deadpool::managed::Pool<AsyncDieselConnectionManager<AsyncPgConnection>>` it doesnt throw error, but i cannot use this variable later on:

![image](https://github.com/weiznich/diesel_async/assets/26543876/c56c0131-6ac9-48a7-ae6b-d60764da11d4)

**Comment by weiznich** [comment]
Closed because it seems like you've just mixed up different types with the name `Pool`. 
For future similar questions: Please use the support forum as the issue tracker is for tracking bugs.

**Comment by mpiorowski** [comment]
@weiznich Yes, sorry, i was playing with different pool providers, so there might be some misunderstanding. But I still believe something is not right here. Cleaned code to show the problem.

```
pub struct MyService {
    // This one is throwing the error
    // pool: diesel_async::pooled_connection::deadpool::Pool<
    //     diesel_async::pooled_connection::AsyncDieselConnectionManager<
    //         diesel_async::AsyncPgConnection,
    //     >,
    // >,
    // this one require me to install deadpool cargo seperate
    pool: deadpool::managed::Pool<
        diesel_async::pooled_connection::AsyncDieselConnectionManager<
            diesel_async::AsyncPgConnection,
        >,
    >,
}
``` 

![image](https://github.com/weiznich/diesel_async/assets/26543876/ddced692-2f13-40d6-96a9-d4092eacc53c)

**Comment by sergey-suslov** [comment]
I have the same issue
```toml
[dependencies]
diesel = { version = "2.1.0", features = [
  "postgres",
  "r2d2",
  "uuid",
  "chrono",
] }
diesel-async = { version = "0.3.1", features = ["postgres", "bb8"] }

```

```rust
use diesel_async::pooled_connection::bb8::Pool;
use diesel_async::pooled_connection::AsyncDieselConnectionManager;
use diesel_async::AsyncPgConnection;

pub type AsyncPool =
    diesel_async::pooled_connection::bb8::Pool<AsyncDieselConnectionManager<AsyncPgConnection>>;

#[derive(Clone)]
pub struct AccountRepository {
    db: AsyncPool,
}

```

```
error[E0277]: the trait bound `AsyncDieselConnectionManager<AsyncPgConnection>: PoolableConnection` is not satisfied
  --> crates/shared/src/lib.rs:10:9
   |
10 |     db: AsyncPool,
   |         ^^^^^^^^^ the trait `PoolableConnection` is not implemented for `AsyncDieselConnectionManager<AsyncPgConnection>`
```

**Comment by mpiorowski** [comment]
The solution here is to diesel_async::pooled_connection::bb8::Pool<AsyncPgConnection>

**Comment by weiznich** [comment]
Please do not use the issue tracker to discuss support issues. It's a place for technical discussions about features and bugs. It's not meant to be a support forum.
